migration-engine/connectors/sql-migration-connector/src/sql_schema_differ/table.rs

```diff
@@ -157,10 +157,16 @@ impl<'schema, 'b> TableDiffer<'schema, 'b> {
 
 /// Compare two SQL indexes and return whether they only differ by name.
 fn indexes_match(first: &IndexWalker<'_>, second: &IndexWalker<'_>) -> bool {
-    let left_cols = first.column_names();
-    let right_cols = second.column_names();
+    let left_cols = first.columns();
+    let right_cols = second.columns();
 
     left_cols.len() == right_cols.len()
```

Comment:
This needs to start diffing on length and sort differences. I'm pretty sure those will only be populated when the preview flag is set, but we should test that.